screens/HomeScreen.js

```diff
@@ -38,11 +39,72 @@ export default class HomeScreen extends React.Component {
     header: null
   };
 
-  selectLandmark = data => {
+  fetchLandmarkPlaceId(address) {
+    const formattedLandmarkAddress = this.formatLandmarkText(address);
+
+    return fetch(
+      `https://maps.googleapis.com/maps/api/place/findplacefromtext/json?input=${formattedLandmarkAddress}&inputtype=textquery&key=${Constants.google.apiKey}`
+    )
+      .then(res => {
+        return res.json();
+      })
+      .then(data => {
+        //grab the first result that the search returns and
+        //get its place id
+        if (data.candidates[0]) {
+          return data.candidates[0].place_id;
+        } else {
+          return "none";
+        }
+      })
+      .catch(error => {
+        console.error(error);
+      });
+  }
+
+  fetchLandmarkDetails(googlePlaceId) {
+    return fetch(
+      `https://maps.googleapis.com/maps/api/place/details/json?place_id=${googlePlaceId}&fields=opening_hours,formatted_phone_number,formatted_address&key=${Constants.google.apiKey}`
+    )
+      .then(responseDetails => {
+        return responseDetails.json();
+      })
+      .then(placeDetails => {
+        return placeDetails.result;
+      })
+      .catch(error => {
+        console.error(error);
+      });
+  }
+
+  formatLandmarkText(text) {
+    let words = text.split(" ");
+    return words.join("%20");
+  }
+
+  async selectLandmark(data) {
+    let placeId = data.placeId;
+    let selectedLandmark = { ...data };
+
+    if (!placeId) {
+      //Fail safe for if we don't have the placeId in the db
```

Comment:
ohhh now i see why you have that method up there. since we currently fetch all of the landmarks when the app loads i don't see a situation in which we would need this, but i'm curious to hear your thinking.

Reply:
Yeah, that makes sense. I sort of coded out both options: this is if we decided not to add all the placeIds to the database and kept the data how it is now. But if we're good with that, I can remove this.